src-plugins/medBinaryOperation/medBinaryOperatorBase.h

```diff
@@ -0,0 +1,54 @@
+/*=========================================================================
+
+ medInria
+
+ Copyright (c) INRIA 2013. All rights reserved.
+ See LICENSE.txt for details.
+ 
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#pragma once
+
+#include <dtkCore/dtkSmartPointer.h>
+#include <medAbstractProcess.h>
+#include "medBinaryOperationPluginExport.h"
+#include "itkImage.h"
+
+class medBinaryOperatorBasePrivate;
+
+class MEDBINARYOPERATIONPLUGIN_EXPORT medBinaryOperatorBase : public medAbstractProcess
+{
+    Q_OBJECT
+    
+public:
+    medBinaryOperatorBase();
+    virtual ~medBinaryOperatorBase();
+
+    template <class ImageType> int run();
+    template <class ImageType, class ImageType2> int runProcess();
+
+private:
+    template <class ImageType> itk::Image<unsigned char, 3>::Pointer castToUChar3(medAbstractData* input);
```

Comment:
If you want to avoid having to include itkImage.h in the header file there are a couple solutions. The easiest here would be to declare the function in an anonymous namespace in the .cpp file.